pkg/utils/kubernetes/health/deployment.go

```diff
@@ -135,16 +135,26 @@ func IsDeploymentUpdated(reader client.Reader, deployment *appsv1.Deployment) fu
 		}
 
 		// Now there might be still pods in the system belonging to an older ReplicaSet of the Deployment.
-		podList := &metav1.PartialObjectMetadataList{}
-		podList.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("PodList"))
-		if err := reader.List(ctx, podList, client.InNamespace(deployment.Namespace), client.MatchingLabels(deployment.Spec.Selector.MatchLabels)); err != nil {
+		exactNumberOfPods, err := DeploymentHasExactNumberOfPods(ctx, reader, deployment)
+		if err != nil {
 			return retry.SevereError(err)
 		}
-
-		if int32(len(podList.Items)) != pointer.Int32Deref(deployment.Spec.Replicas, 1) {
+		if !exactNumberOfPods {
 			return retry.MinorError(errors.New("there are still non-terminated old pods"))
 		}
 
 		return retry.Ok()
 	}
 }
+
+// DeploymentHasExactNumberOfPods returns true when there are exactly as many pods as the .spec.replicas field of the
+// deployment mandates.
+func DeploymentHasExactNumberOfPods(ctx context.Context, reader client.Reader, deployment *appsv1.Deployment) (bool, error) {
+	podList := &metav1.PartialObjectMetadataList{}
+	podList.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("PodList"))
+	if err := reader.List(ctx, podList, client.InNamespace(deployment.Namespace), client.MatchingLabels(deployment.Spec.Selector.MatchLabels)); err != nil {
+		return false, err
+	}
+
+	return int32(len(podList.Items)) == pointer.Int32Deref(deployment.Spec.Replicas, 0), nil
```

Comment:
```suggestion
	return int32(len(podList.Items)) == pointer.Int32Deref(deployment.Spec.Replicas, 1), nil
```
Because `replicas` default to 1 in deployments.